portals/devportal/src/main/webapp/source/src/app/components/Apis/Details/Documents/GenerateDocument.jsx

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright (c) 2022, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import React, { useState, useContext, useEffect } from 'react';
+import { withStyles } from '@material-ui/core/styles';
+import { API } from '@stoplight/elements';
+import './elements.css';
+import { ApiContext } from 'AppComponents/Apis/Details/ApiContext';
+import YAML from 'js-yaml';
+import Api from 'AppData/api';
+
+const styles = () => ({
+    generatedDocument: {
+        width: '100%',
+        margin: 50,
+    },
+});
+
+function GenerateDocument(props) {
+    const { classes } = props;
+    const { api } = useContext(ApiContext);
+    const [swagger, updateSwagger] = useState('');
+
+    useEffect(() => {
+        const apiClient = new Api();
+        const promisedApi = apiClient.getSwaggerByAPIId(api.id);
+        promisedApi
+            .then((response) => {
+                updateSwagger(YAML.safeDump(YAML.safeLoad(response.data)));
+            })
+            .catch((error) => {
+                if (process.env.NODE_ENV !== 'production') {
+                    console.log(error);
```

Comment:
Will this be a console.log() or console.error(). Please check

Reply:
[PR#425](https://github.com/wso2/apim-apps/pull/425) fixes this.